ivy_tests/test_ivy/test_frontends/test_tensorflow/test_math.py

```diff
@@ -1382,6 +1382,36 @@ def test_tensorflow_unsorted_segment_sqrt_n(
     )
 
 
+# unsorted_segment_prod
+@handle_frontend_test(
+    fn_tree="tensorflow.math.unsorted_segment_prod",
+    data=helpers.array_values(dtype=ivy.int32, shape=(5, 6), min_value=1, max_value=9),
+    segment_ids=helpers.array_values(
+        dtype=ivy.int32, shape=(5,), min_value=0, max_value=4
+    ),
+    test_with_out=st.just(False),
+)
+def test_tensorflow_unsorted_segment_prod(
+    *,
+    data,
+    segment_ids,
+    frontend,
+    test_flags,
+    fn_tree,
+    on_device,
+):
+    helpers.test_frontend_function(
+        input_dtypes=[ivy.float32, ivy.int32],
```

Comment:
`input_dtypes` should not be hardcoded. You should be passing the `data` and `segment_ids` dtypes from `@handle_frontend_test`.